### PR TITLE
chore: tidy up the flow

### PR DESCRIPTION
**Pull request by Decodetalkers**
1. check the value when dconfig changed, this will make set not do twice
    2. write config file before write to dconfig. Because dde-desktop listen
             to dconfig, so we should update config before dconfig changed. And
             base config need to be changed before write new dconfig

### CODE REVIEW

**Comment by Decodetalkers** [comment]
- https://github.com/linuxdeepin/developer-center/issues/6356
- https://github.com/linuxdeepin/developer-center/issues/6338

**Review by kegechen — COMMENTED** [review]
+1

File: src/service/impl/appearancemanager.cpp [hunk 1]
```diff
@@ -34,7 +34,30 @@
 
 DCORE_USE_NAMESPACE
 
-const QString wallpaperJsonPath = QString("%1/dde-appearance/").arg(utils::GetUserConfigDir());
+const static QString wallpaperJsonPath = QString("%1/dde-appearance/").arg(utils::GetUserConfigDir());
+
+static void writeWallpaperConfig(const QVariant &wallpaper)
```
Comment (dengbo11) [inline src/service/impl/appearancemanager.cpp:39]: 这个函数放到modules/api/utils.h中？

Comment (Decodetalkers) [reply src/service/impl/appearancemanager.cpp]: 好

**Review by dengbo11 — APPROVED** [review]

**Pull request merged**